src/api/firebase-auth.ts

```diff
@@ -0,0 +1,18 @@
+import {
+    createUserWithEmailAndPassword,
+    getAuth,
+    signInWithEmailAndPassword,
+    signOut as fSignOut,
+    onAuthStateChanged,
+    NextOrObserver,
+    User,
+} from 'firebase/auth';
+
+import { firebaseApp } from '../providers/firebase/firebase';
+import { Creds } from '../utils/types';
```

Comment:
```suggestion
import type { Creds } from '../utils/types';
```
можно, если только типы забираешь. И можно сокращения не юзать